galaxykit/client.py

```diff
@@ -187,6 +187,15 @@ def _http(self, method, path, *args, **kwargs):
                 method, url, headers=headers, verify=self.https_verify, *args, **kwargs
             )
 
+        elif "Invalid token" in resp.text:
+            # If invalid token, let's retry with basic auth token
```

Comment:
Can we also include some explanation of _why_ we want to do this? What is the situation where this happens and it makes sense to (potentially?) change the authentication type?

Reply:
now that we've got a lot of user tests I've found many 'invalid token' errors, this refresh-and-retry mechanism seems to work well. The authentication type change is just for simplicity, only to avoid calls to auth_url and getting the token. But If you think we should stick to the original authentication type I'll try to do it

Reply:
@ironfroggy  I still use the refresh and retry mechanism since the tests that change user details seem to invalidate the token but I'm not using basic authentication anymore, I use the same one as the tests use, getting the token via v3/auth/token/ (token_type = 'Token' and not 'Basic')